src/libcore/str/pattern.rs

```diff
@@ -47,6 +47,22 @@ pub trait Pattern<'a>: Sized {
         matches!(self.into_searcher(haystack).next(), SearchStep::Match(0, _))
     }
 
+    /// Removes the pattern from the front of haystack, if it matches.
+    #[inline]
+    fn strip_prefix_from(self, haystack: &'a str) -> Option<&'a str> {
```

Comment:
I like the 
```suggestion
    fn strip_prefix_of(self, haystack: &'a str) -> Option<&'a str> {
```
better for consistence with `is_prefix_of`

Reply:
Fine, done.